packages/core/src/sandbox/linux/landlock-runner.rs

```diff
@@ -0,0 +1,216 @@
+// Copyright 2025 Google LLC
+// SPDX-License-Identifier: Apache-2.0
+
+#![allow(clippy::missing_safety_doc)]
+
+use enumflags2::BitFlags;
+use landlock::{
+    Access, AccessFs, CompatLevel, Compatible, LandlockStatus, PathBeneath, PathFd, Ruleset,
+    RulesetAttr, RulesetCreated, RulesetCreatedAttr, RulesetStatus, ABI,
+};
+use libc::{c_char, execvp};
+use std::env;
+use std::ffi::{CString, OsString};
+use std::os::unix::ffi::OsStringExt;
+use std::path::{Path, PathBuf};
+use std::process::exit;
+
+fn main() {
+    let mut args = env::args_os().collect::<Vec<_>>();
+    if args.len() < 2 {
+        usage_and_exit();
+    }
+
+    // Strip program name
+    args.remove(0);
+
+    let mut rw_paths: Vec<OsString> = Vec::new();
+    let mut ro_paths: Vec<OsString> = Vec::new();
+
+    while !args.is_empty() {
+        if args[0] == "--" {
+            args.remove(0);
+            break;
+        }
+        if args.len() < 2 {
+            usage_and_exit();
+        }
+        let flag = args.remove(0);
+        let path = args.remove(0);
+        match flag.as_os_str().to_str() {
+            Some("--rw") => rw_paths.push(path),
+            Some("--ro") => ro_paths.push(path),
+            _ => usage_and_exit(),
+        }
+    }
+
+    if args.is_empty() {
+        eprintln!("[landlock-runner] Missing command to execute");
+        exit(64);
+    }
+
+    // Request the latest ABI the crate knows about; compatibility will fall back when needed.
+    let abi = ABI::V6;
+    let ro_access = AccessFs::from_read(abi);
+    let rw_access = AccessFs::from_all(abi);
+
+    let handled = ro_access | rw_access;
+
+    let mut ruleset = match Ruleset::default()
+        .set_compatibility(CompatLevel::BestEffort)
+        .handle_access(handled)
+    {
+        Ok(ruleset) => match ruleset.create() {
+            Ok(created) => created,
+            Err(err) => {
+                eprintln!("[landlock-runner] landlock_create_ruleset failed ({err})");
+                exit(112);
+            }
+        },
+        Err(err) => {
+            eprintln!("[landlock-runner] landlock_create_ruleset failed ({err})");
+            exit(112);
+        }
+    };
+
+    for p in ro_paths {
+        let (path, access) = normalize_path_and_access(p, false, abi);
+        ruleset = add_path_rule(ruleset, path, access);
+    }
+    for p in rw_paths {
+        let (path, access) = normalize_path_and_access(p, true, abi);
+        ruleset = add_path_rule(ruleset, path, access);
+    }
+
+    let status = match ruleset.restrict_self() {
+        Ok(status) => status,
+        Err(err) => {
+            eprintln!("[landlock-runner] landlock_restrict_self failed ({err})");
+            exit(112);
+        }
+    };
+
+    if matches!(
+        status.landlock,
+        LandlockStatus::NotEnabled | LandlockStatus::NotImplemented
```

Comment:
There is no need to check `status.landlock`, only `status.ruleset` is useful in this case. `status.landlock` is only useful to log details.